### ISSUE TITLE
Remove API keys and other sensitive data from *Model.Builder

### ISSUE THREAD

**Issue by ghost** [langchain4j/langchain4j#2139]
Make sure API keys and other sensitive data is not logged in `toString()` of models and model builders.

Lombok generates `toString` for `@Builder` which leads to logging of API keys and other sensitive data that should not be exposed via `toString`.

**Comment by remyohajinwa** [comment]
Can I pick this one @langchain4j

**Comment by langchain4j** [comment]
@remyohajinwa sure, but how would you go about it? Currently, lombok is still used in many places, I guess it would make sense to do this issue once we fully get rid of lombok?

**Comment by remyohajinwa** [comment]
Yeah, that would make more sense to wait until we get rid of lombok. Guess I will pick another issue

**Comment by sergioverde90** [comment]
Any approach to implement this? I came up with the idea of implementing a unit test to verify that for each class implementing the *Model interface (using reflection) the toString() method does not contain the word apiKey or similar. What do you think?
